packages/next-on-pages/src/buildApplication/processVercelFunctions/ast.ts

```diff
@@ -2,58 +2,69 @@ import type { Node } from 'acorn';
 import type * as AST from 'ast-types/gen/kinds';
 import type { ProcessVercelFunctionsOpts } from '.';
 import { cliError } from '../../cli';
+import { createHash } from 'crypto';
 
 /**
  * Collects the identifiers from the AST and adds them to the provided maps and lists.
  *
  * @param collectedIdentifiers The collected identifiers.
- * @param program The AST for the function file.
+ * @param programInfo The AST for the function file and its entrypoint.
  * @param opts The options for the function processing.
  */
 export function collectIdentifiers(
 	{ identifierMaps, programIdentifiers }: CollectIdentifiersOpts,
-	program: AST.ProgramKind,
+	programInfo: ProgramInfo,
 	{ disableChunksDedup }: Partial<ProcessVercelFunctionsOpts>,
 ): void {
 	// Wasm
 	collectIdentifiersByType(
 		{ programIdentifiers, foundIdentifiers: identifierMaps.wasm },
-		program,
+		programInfo,
 		extractWasm,
 	);
 
 	// Manifests
 	collectIdentifiersByType(
 		{ programIdentifiers, foundIdentifiers: identifierMaps.manifest },
-		program,
+		programInfo,
 		extractManifest,
 	);
 
 	// Webpack chunks
 	if (!disableChunksDedup) {
 		collectIdentifiersByType(
 			{ programIdentifiers, foundIdentifiers: identifierMaps.webpack },
-			program,
+			programInfo,
 			extractWebpack,
 		);
 	}
 }
 
+/**
+ * Groups the identifiers in the provided maps by their consumers.
+ *
+ * @param identifierMaps The maps of found identifiers in the AST.
+ */
+export function groupIdentifiers(identifierMaps: IdentifierMaps) {
+	Object.values(identifierMaps).forEach(map => groupIdentifiersByType(map));
```

Comment:
sorry after reviewing better the code I think you had it right the first time around 😓
```suggestion
	// Webpack Chunks
	groupIdentifiersByType(identifierMaps.webpack);
	// Manifests
	groupIdentifiersByType(identifierMaps.manifest);
```